File: server/src/models/recipes/RecipeSearchModel.js
import RecipeModel from "./RecipeModel.js";

class RecipeSearchModel {
  constructor({ page_number = null, recipes = null } = {}) {
    this.page_number = page_number;
    this.recipes = recipes;
  }

  static fromJson(json) {
    const recipesJson = json["recipes"];
    const recipeJson = recipesJson ? recipesJson["recipe"] : null;
    const recipes = recipeJson?.map(e => RecipeModel.fromJson(e)) ?? [];

    return new RecipeSearchModel({
      page_number: recipesJson?.["page_number"] ? parseInt(recipesJson["page_number"]) : null,
      recipes: recipes,
    });
  }
}

export default RecipeSearchModel;

File: lib/data/models/recipes/recipe_search_model.dart
import 'package:calory_tool/data/models/recipes/recipe_model.dart';
import 'package:hive_flutter/hive_flutter.dart';

part 'recipe_search_model.g.dart';

@HiveType(typeId: 12)
class RecipeSearchModel extends HiveObject {
  RecipeSearchModel({required this.recipes, this.pageNumber});

  factory RecipeSearchModel.fromJson(Map<String, dynamic> json) {
    return RecipeSearchModel(
      pageNumber: (json['page_number'] as num?)?.toInt(),
      recipes:
          (json['recipes'] as List?)
              ?.map((e) => RecipeModel.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }

  @HiveField(0)
  final int? pageNumber;

  @HiveField(1)
  final List<RecipeModel> recipes;
}
